src/apps/answers/service.py

```diff
@@ -568,6 +581,48 @@ async def get_activity_answer(
         )
 
         return submission
+    
+    async def _validate_relation_between_subjects_in_applet(
+            self,
+            respondent_subject: SubjectSchema,
+            target_subject: SubjectSchema,
+            source_subject: SubjectSchema
+    ) -> None:
+        
+        """
+            Validate the relationship between subjects in an applet.
+            - respondent_subject: the subject inputting the answers.
+            - target_subject: the subject about whom the responses are.
+            - source_subject: the subject providing the responses.
+            
+            This method checks:
+            1. If there is a relationship between the logged-in user (respondent_subject) and target_subject.
+            2. If there is a relationship between the respondent_subject and the source_subject.
+            Raises:
+                MultiinformantAssessmentNoAccessApplet: If no valid relationship is found.
+        """
+
+        if respondent_subject.id == target_subject.id or respondent_subject.id == source_subject.id:
+            return None
+
+        relation_respondent_target_subjects_call = SubjectsCrud(self.session).get_relation(
+            respondent_subject.id, target_subject.id
+        )
```

Comment:
there call returns a coroutine, we usually name this var as relation_respondent_target_subjects_coro, but it is only a sugestion